tests/Insightly/Listeners/ActivateIntegrationWithCouponTest.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Insightly\Listeners;
+
+use App\Domain\Coupons\Coupon;
+use App\Domain\Coupons\Repositories\CouponRepository;
+use App\Domain\Integrations\Events\IntegrationActivatedWithCoupon;
+use App\Domain\Integrations\Integration;
+use App\Domain\Integrations\IntegrationStatus;
+use App\Domain\Integrations\IntegrationType;
+use App\Domain\Integrations\Repositories\IntegrationRepository;
+use App\Insightly\InsightlyMapping;
+use App\Insightly\Listeners\ActivateIntegrationWithCoupon;
+use App\Insightly\Objects\OpportunityStage;
+use App\Insightly\Objects\OpportunityState;
+use App\Insightly\Repositories\InsightlyMappingRepository;
+use App\Insightly\Resources\ResourceType;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Tests\MockInsightlyClient;
+
+final class ActivateIntegrationWithCouponTest extends TestCase
+{
+    use MockInsightlyClient;
+
+    private ActivateIntegrationWithCoupon $listener;
+
+    private IntegrationRepository&MockObject $integrationRepository;
+
+    private InsightlyMappingRepository&MockObject $insightlyMappingRepository;
+
+    private CouponRepository&MockObject $couponRepository;
+
+    private UuidInterface $integrationId;
+
+    private string $couponCode;
+
+    protected function setUp(): void
+    {
+        $this->integrationId = Uuid::uuid4();
+        $this->couponCode = 'test123';
+        $this->integrationRepository = $this->createMock(IntegrationRepository::class);
+        $this->insightlyMappingRepository = $this->createMock(InsightlyMappingRepository::class);
+        $this->couponRepository = $this->createMock(CouponRepository::class);
+
+        $this->mockCrmClient();
+
+        $this->listener = new ActivateIntegrationWithCoupon(
+            $this->insightlyClient,
+            $this->integrationRepository,
+            $this->insightlyMappingRepository,
+            $this->couponRepository,
+            $this->createMock(LoggerInterface::class),
+        );
+    }
+
+    public function test_it_creates_a_project_when_activating_with_a_coupon(): void
+    {
+        // Given
+        $insightlyOpportunityId = 42;
+        $insightlyProjectId = 51;
+        $integration = $this->givenThereIsAnIntegrationWithId($this->integrationId);
+        $this->givenThereIsAnOpportunityMapping($this->integrationId);
+        $coupon = $this->givenThereIsACoupon($this->integrationId);
+
+        $this->integrationRepository->activateWithCouponCode($this->integrationId, $this->couponCode);
```

Comment:
There is no need to call this method `activateWithCouponCode` on a mock.

Reply:
Removed in https://github.com/cultuurnet/publiq-platform/pull/111/commits/a60d00a37afe05f53dbbca1607d9abfbe8cb36ab